metals/src/main/scala/scala/meta/internal/metals/PackageProvider.scala

```diff
@@ -7,25 +7,43 @@ import org.eclipse.lsp4j.WorkspaceEdit
 
 import scala.meta.internal.metals.MetalsEnrichments._
 import scala.meta.io.AbsolutePath
+import java.nio.file.Path
 
 class PackageProvider(private val buildTargets: BuildTargets) {
 
   def workspaceEdit(path: AbsolutePath): Option[WorkspaceEdit] = {
-    packageName(path).map(workspaceEdit(path, _))
+    packageStatement(path).map(workspaceEdit(path, _))
   }
 
-  private def packageName(path: AbsolutePath): Option[String] = {
-    // package.scala file needs to be handled separately
-    // as its package name is different than package name of normal scala file
-    if (path.isScala &&
-      path.toFile.length() == 0 &&
-      path.filename != "package.scala") {
-      val sourceItem = buildTargets.inverseSourceItem(path)
-      sourceItem
+  private def packageStatement(path: AbsolutePath): Option[String] = {
+
+    def packageObjectStatement(path: Iterator[Path]): Option[String] = {
+      val pathList = path.toList
+      val packageDeclaration =
+        if (pathList.size > 1)
+          s"package ${pathList.dropRight(1).mkString(".")}\n"
+        else ""
+      pathList.lastOption.map { packageObjectName =>
+        s"""|${packageDeclaration}package object $packageObjectName{
```

Comment:
Isn't there usually a newline in between the `package` and the `package object`. I mean something like

```scala
package a.b

package object c {

}
```

Reply:
Fixed